examples/kyb-app/src/components/organisms/DynamicUI/PageResolver/PageResolver.tsx

```diff
@@ -0,0 +1,25 @@
+import { useCurrentPageElement } from '@app/components/organisms/DynamicUI/PageResolver/hooks/useCurrentPageElement';
+import {
+  PageResolverContext,
+  PageResolverProps,
+} from '@app/components/organisms/DynamicUI/PageResolver/types';
+import { pageResolverContext } from './page-resolver.context';
+import { useMemo } from 'react';
+
+const { Provider } = pageResolverContext;
+
+export const PageResolver = ({ state, pages, children }: PageResolverProps) => {
+  const currentPage = useCurrentPageElement(state, pages);
+
+  const context: PageResolverContext = useMemo(
+    () => ({ currentPage, pages }),
+    [currentPage, pages],
+  );
+
+  const child = useMemo(
+    () => (typeof children === 'function' ? children(context) : children),
+    [children, context],
+  );
```

Comment:
See previous comment about `useMemo`.

Reply:
You found good point and this one doesnt make sense in way how it is used right now. I will get rid of this when DynamicUI will start refactor/rework using hooks instead of multiple providers.